pkgs/by-name/ed/edopro/package.nix

```diff
@@ -0,0 +1,262 @@
+    };
+
+    nativeBuildInputs = [ premake5 ];
+
+    buildInputs = [
+      lua5_3
+    ];
+
+    preBuild = ''
+      premake5 gmake2
+    '';
+
+    enableParallelBuilding = true;
+    buildFlags = "verbose=true config=release ocgcore";
+    makeFlags = "-C build";
+
+    installPhase = ''
+      mkdir -p $out/bin
+      cp bin/release/libocgcore.a $out/bin
+    '';
+  };
+
+  edopro = stdenv.mkDerivation rec {
+    pname = "edopro";
+    version = edopro-version;
+
+    src = fetchFromGitHub {
+      owner = "edo9300";
+      repo = pname;
+      rev = "c713e23491a1e55c9d8e91257e5f2b5873696b9b";
+      hash = "sha256-2E1cjG0FONu/fbr67/3qRCKQ1W/wPznClEWsMa1FAzo=";
+      fetchSubmodules = true;
+    };
+
+    nativeBuildInputs = [ premake5 ];
+
+    buildInputs = [
+      bzip2
+      curl
+      flac
+      fmt_9
+      freetype
+      irrlicht-edopro
+      libevent
+      libgit2
+      libjpeg
+      libpng
+      libvorbis
+      nlohmann_json
+      openal
+      SDL2
+      sqlite
+    ];
+
+    # nixpkgs' gcc stack currently appears to not support LTO
+    postPatch = ''
+      sed -i '/LinkTimeOptimization/d' ./premake5.lua
+    '';
+
+    preBuild = ''
+      premake5 gmake2 \
+        --bundled-font="${font}" \
+        --covers=\"${covers_url}\" \
+        --fields=\"${fields_url}\" \
+        --no-core \
+        --no-joystick=false \
+        --pics=\"${pics_url}\" \
+        --prebuilt-core="${ocgcore}/bin" \
+        --sound=sfml
+    '';
+
+    enableParallelBuilding = true;
+    CXXFLAGS = "-include cstdint";
+    LDFLAGS = "-I ${irrlicht-edopro}/include -L ${irrlicht-edopro}/bin";
+    buildFlags = "verbose=true config=release_x64 ygoprodll";
+    makeFlags = "-C build";
+
+    # TODO find out how to get arch name from nix for cross compiled
+    installPhase = ''
+      mkdir -p $out/bin
+      cp bin/x64/release/ygoprodll $out/bin
+    '';
+  };
+
+  edopro-script =
+    let
+      assetsToCopy = lib.concatStringsSep "," [
+        # Needed if we download files from ProjectIgnis' sercet website
+        # "cacert.pem"
+        "config"
+        "deck"
+        "COPYING.txt"
+        "expansions"
+        "lflists"
+        "notices"
+        "puzzles"
+        "fonts"
+        "script"
+        "skin"
+        "sound"
+        "textures"
+        "WindBot"
+      ];
+    in
+    writeShellScriptBin "EDOPro" ''
+      set -eu
+      EDOPRO_DIR="$HOME/.local/share/edopro"
+
+      if [ ! -d $EDOPRO_DIR ]; then
+          mkdir -p $EDOPRO_DIR
+          cp -r ${assets.src}/{${assetsToCopy}} $EDOPRO_DIR
+
+          find $EDOPRO_DIR -type d -exec chmod u=rwx,go-rwx {} +
+          find $EDOPRO_DIR -type f -exec chmod u=rw,go-rwx {} +
+
+          rm $EDOPRO_DIR/config/io.github.edo9300.EDOPro.desktop.in
+      fi
+
+      export LD_LIBRARY_PATH=$EDOPRO_DIR:'${
+        lib.makeLibraryPath [
+          libGL
+          libX11
+          libxkbcommon
+          libXxf86vm
+          wayland
+          egl-wayland
```

Comment:
Sound isn't working on my end currently; I imagine the difference will be that one of us is using pipewire and the other pulse (or raw alsa, I guess).

If you scroll through the linux help channel on the discord you'll hopefully see some comments from me explaining how to get sound working after deliberation with the maintainer; basically, both pipewire and pulse's libraries need to be available for sfml to know how to play audio.

I'd dig it up for you, but I woke up kickbanned the morning after getting the package done - probably because I asked for help trying to package this for NixOS, though the maintainer seemed mostly amicable up until that point, so I'm still confused about that.